localstack/services/cloudformation/models/iam.py

```diff
@@ -327,6 +328,35 @@ def get_deploy_templates(cls):
         }
 
 
+class IAMServiceLinkedRole(GenericBaseModel):
+    @staticmethod
+    def cloudformation_type():
+        return "AWS::IAM::ServiceLinkedRole"
+
+    def fetch_state(self, stack_name, resources):
+        iam = aws_stack.connect_to_service("iam")
+        roles = iam.list_roles(MaxItems=100)["Roles"]
```

Comment:
There will be case where this leads to issues without proper paging and getting all roles.

*But* I think we don't even need to go this route if we restructure the way we apparently use the service linked roles in localstack.


AWS automatically creates a proper Path for the role so we could simply filter by the path prefix in list_roles.

On AWS you can do: `aws iam list-roles --path-prefix "/aws-service-role"` to get a list of all linked service roles or for a single one:  `aws iam list-roles --path-prefix "/aws-service-role/eks.amazonaws.com"`

Reply:
Perfect, good point 👍 . Was already scratching my head if the name prefix is the best approach here. Will refactor..